helix-term/src/commands.rs

```diff
@@ -1668,6 +1668,36 @@ mod cmd {
         Ok(())
     }
 
+    fn buffer_close(
```

Comment:
Let's extract this onto a function on the `Editor`, that way it can be used elsewhere.

Reply:
That should also make it easier to extend in the future, in case there are more "cleanup" actions that need to be taken (e.g. more issues like the below language server one appear).